state.go

```diff
@@ -317,6 +292,66 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string) (bool, time.Duration, error) {
```

Comment:
I think it would be cleaner if Ping and pingNode returned duration and error.  `error != nil` should mean the same thing as `success == false`, so there would be no need for the success return value.